### PR TITLE
OP-1187 | Add function for printing inventory report

### PR DESCRIPTION
**Pull request by JantBogard**
See [OP-1187](https://openhospital.atlassian.net/browse/OP-1187).

[OP-1187]: https://openhospital.atlassian.net/browse/OP-1187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by dbmalkovsky** [comment]
Please start with a more recent branch to add this functionality.   432 changed files is out of line.   Besides there are multiple conflicts.   No one can review this PR the way it is.

**Review by dbmalkovsky — COMMENTED** [review]
Please start with a more recent branch to add this functionality. 432 changed files is out of line. Besides there are multiple conflicts. No one can review this PR the way it is.

**Comment by JantBogard** [comment]
> Please start with a more recent branch to add this functionality. 432 changed files is out of line. Besides there are multiple conflicts. No one can review this PR the way it is.

Sorry, I forgot to update this branch with the recently develop branch. I think now it's ok

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 1]
```diff
@@ -1097,4 +1098,32 @@ private static Date toDate(LocalDate localDate) {
 		return Optional.ofNullable(localDate).map(t -> Date.from(t.atStartOfDay().atZone(ZoneId.systemDefault()).toInstant())).orElse(null);
 	}
 
-}
+	public JasperReportResultDto getInventoryReportPdf(MedicalInventory inventory, String jasperFileName, int printQtyReal) throws OHServiceException {
+		try {
+			Hospital hosp = hospitalManager.getHospital();
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java]: This line isn't needed.  the `getHospitalParameters()` call in line 1105 get the hospital object isself.

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 2]
```diff
@@ -1097,4 +1098,32 @@ private static Date toDate(LocalDate localDate) {
...
+			parameters.put("inventoryStatus", status != null ? status : inventory.getStatus());
+			parameters.put("printQtyReal", printQtyReal);
+
+			LocalDateTime inventoryDateTime = inventory.getInventoryDate();
+			DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DD_MM_YYYY);
+			String formattedDate = inventoryDateTime.format(formatter);
+			parameters.put("inventoryDate", formattedDate);
+
+			String pdfFilename = compilePDFFilename(RPT_BASE, jasperFileName, Arrays.asList(String.valueOf(inventory.getId())), "pdf ");
+			LOGGER.info("Generated PDF File: " + pdfFilename);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java]:
```suggestion
			LOGGER.info("Generated PDF File: {}", pdfFilename);
```

**Review by dbmalkovsky — COMMENTED** [review]
left some comments

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 3]
```diff
@@ -1096,4 +1097,31 @@ private static Date toDate(LocalDate localDate) {
...
+			JasperExportManager.exportReportToPdfFile(result.getJasperPrint(), pdfFilename);
+
+			return result;
+		} catch (Exception e) {
+			LOGGER.error("", e);
+			throw new OHServiceException(e, new OHExceptionMessage(MessageBundle.getMessage(STAT_REPORTERROR_MSG)));
+		}
+	}
+
+}
```
Comment (mwithi) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java]:
```suggestion
	public JasperReportResultDto getInventoryReportPdf(MedicalInventory inventory, String jasperFileName, int printRealQty) throws OHServiceException {
		try {
			String status = inventory.getStatus();
			Map<String, Object> parameters = new HashMap<>(getHospitalParameters());
			parameters.put("inventoryId", String.valueOf(inventory.getId()));
			parameters.put("inventoryReference", inventory.getInventoryReference());
			parameters.put("inventoryStatus", status != null ? status : inventory.getStatus());
			parameters.put("printRealQty", printRealQty);

			LocalDateTime inventoryDateTime = inventory.getInventoryDate();
			DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DD_MM_YYYY);
			String formattedDate = inventoryDateTime.format(formatter);
			parameters.put("inventoryDate", formattedDate);

			String pdfFilename = compilePDFFilename(RPT_BASE, jasperFileName, Arrays.asList(String.valueOf(inventory.getId())), "pdf ");
			LOGGER.info("Generated PDF File: {}", pdfFilename);

			JasperReportResultDto result = generateJasperReport(compileJasperFilename(RPT_BASE, jasperFileName), pdfFilename, parameters);
			JasperExportManager.exportReportToPdfFile(result.getJasperPrint(), pdfFilename);

			return result;
		} catch (Exception e) {
			LOGGER.error("", e);
			throw new OHServiceException(e, new OHExceptionMessage(MessageBundle.getMessage(STAT_REPORTERROR_MSG)));
		}
	}

}
```

**Review by mwithi — APPROVED** [review]
LGTM. Thanks a lot!

**Pull request merged**